src/util/Router.ts

```diff
@@ -0,0 +1,87 @@
+import { MovieDetails } from "../abstracts/type";
+import AppComponent from "../components/AppComponent";
+import AppHeaderComponent from "../components/AppHeaderComponent";
+import MovieModalComponent from "../components/modal/MovieModalComponent";
+import { ANIMATED_TIME } from "../constants/constants";
+import { fetchMovieDetails } from "./Api";
+import MovieListComponent from "../components/movie/MovieListComponent";
+
+const routes = {
+  "/": app,
+  "/movie/:id": movieDetail,
+  "/search/:keyword": searchList,
+};
+
+const closeModal = () => {
+  const modal = document.querySelector("movie-modal") as HTMLElement;
+
+  if (modal) {
+    modal.style.opacity = "0";
+    setTimeout(() => {
+      modal.remove();
+    }, ANIMATED_TIME.MODAL);
+  }
+};
+
+async function app() {
+  closeModal();
+
+  const list = document.querySelector("movie-list") as MovieListComponent;
+  await list.popularListInit();
+}
+
+async function movieDetail(id: string) {
+  const modal = document.createElement("movie-modal") as MovieModalComponent;
+  modal.loadMovieDetail(id);
+
+  const app = document.querySelector("#app") as HTMLDivElement;
+  app.append(modal);
+
+  setTimeout(() => {
+    modal.classList.add("fadein");
+  });
+}
+
+async function searchList(keyword: string) {
+  closeModal();
+
+  const app = document.querySelector("app-component") as AppComponent;
```

Comment:
이렇게 사용하고 app이 없을 때의 error 처리를 추가해봐도 좋겠네요

```suggestion
  const app = document.querySelector<AppComponent>("app-component");
```